src/SMTLIB/Backends/Process.hs

```diff
@@ -59,13 +62,14 @@ data Handle = Handle
   }
 
 -- | Run a solver as a process.
+-- Failures relative to terminating the process are logged and discarded.
 new ::
   -- | The solver process' configuration.
   Config ->
   -- | A function for logging the solver's creation, errors and termination.
-  (BS.ByteString -> IO ()) ->
+  (Text -> IO ()) ->
```

Comment:
Why not make the logging function part of the `Config` ?

Reply:
I'm trying to stick with the [Handle design pattern](https://jaspervdj.be/posts/2018-03-08-handle-pattern.html). Since the logger is doing IO it should be part of the `Handle` datatype if anything. But we only use the logger during initialization so it doesn't make much sense to have it around, hence I just have it be a one-off argument (as suggested by the design pattern).

Reply:
It is worth paying attention to the applicable patterns. In this case however, I think there is no good reason to require an extra parameter on every invocation, or adding extra initialization functions for this sake.

> But we only use the logger during initialization so it doesn't make much sense to have it around

I'm not sure what is the meaning of "having it around", but putting the value in the `Config` also looks as a one-off thing, or even a zero-off thing if the default is used.

Reply:
By having it around I meant having it around inside the `Handle` datatype.